feat: Introduce `utils-aws` package

pgrzesik opened the pull request.

mnapoli commented on utils-aws/src/index.js:
```diff
@@ -0,0 +1,9 @@
+
+module.exports = {
+  getCredentialProvider,
+  ensureAwsSpecificEnvironmentVariables,
```
I'm guessing `ensureAwsSpecificEnvironmentVariables()` will be used by Compose (CLI package) at the initialization of the CLI process? (so that way the env variables are "normalized/fixed" globally for all components)

mnapoli commented on utils-aws/src/utils/default-provider.js:
```diff
@@ -0,0 +1,73 @@
+async function defaultProvider(init) {
+  return memoize(
+    chain(
+      // Change the logic to not skip `fromEnv` when process.env.AWS_PROFILE is present
```
Could we detail "why" in the comment? That way we have a bit of background why we do the change, if we ever need to adjust it.

pgrzesik commented:
@mnapoli I've updated a few things:
- Introduced region resolution (it's missing resolution from EC2 metadata service, but it's low priority and I will implement it in the meantime)
- Cleanup the code of `utils-aws` a bit to remove unnecessary util that enforces env vars
- Change how region is resolved in AwsComponent
- Change how region is accessed in CDK class

Please let me know if you have any suggestions to that - as next steps I will work on docs, improving the region resolution and publishing the first version of the library (to unblock finishing state implementation that can use provider credentials resolver)

mnapoli reviewed (approved):
I don't have anything to add, looks great!
